helix-term/src/commands.rs

```diff
@@ -3802,11 +3803,15 @@ pub mod insert {
     }
 
     pub fn insert_tab(cx: &mut Context) {
+        insert_tab_impl(cx, 1)
+    }
+
+    fn insert_tab_impl(cx: &mut Context, count: usize) {
         let (view, doc) = current!(cx.editor);
         // TODO: round out to nearest indentation level (for example a line with 3 spaces should
         // indent by one to reach 4 spaces).
 
-        let indent = Tendril::from(doc.indent_style.as_str());
+        let indent = Tendril::from(doc.indent_style.as_str().repeat(count));
```

Comment:
handling `count` is done here rather than in the new `insert_char` command because it needs to be taken into account if this TODO is ever fixed; putting it outside `insert_tab` makes it more likely to be forgotten.